### DISCUSSION TITLE
How to use keyword argument for custom marker in run configuration (pycharm)?

### DISCUSSION THREAD

**Discussion by mandras73** [pytest-dev/pytest#12980] [General] [answered]
Hi, according to the documentation, we could use custom marker with keyword argument, like
`@pytest.mark.unit(state='new')`

will be executedas

`$ pytest -v -m "unit(state='new')"`

I'm using playwright pytest run configuration and executing the custom marker (from pycharm) with:
`-m=unit`

But don't know how should I add the param to the argument when only would like to execute unit mark where state=new

I also tried
`@pytest.mark.unit('new')`
executed
`-m=unit -k=new`
but didn't work.

Thanks

**Comment by mandras73** [comment]
I tested it and reviewed the updated doc, but still keyword argument only works for me with integer, but not with string.
This works
`@pytest.mark.unit(state=1)`
executed with
`-m=unit(state=1)`

**Reply by The-Compiler** [reply]
Contrary to the example in the docs, you're not using `'...'` or `"..."` around your string.

**Reply by mandras73** [reply]
In the first post, I'm using it, like
`@pytest.mark.unit(state='new')`

**Reply by mandras73** [reply]
tried with
`@pytest.mark.unit(state='new')`
and 
`@pytest.mark.unit(state="new")`
and when tried to execute with different ways, got errors

```
-m=unit(state='new')
-m=unit(state="new")
-m=unit(state=new)
```
ERROR: Wrong expression passed to '-m': unit(state=new): at column 13: unexpected character/s "new"

**Reply by The-Compiler** [reply]
The error clearly shows that you're not passing it with quotes correctly. Your example from the (edited) original post works fine for me.

**Reply by RonnyPfannschmidt** [reply, marked as answer]
As far as u can tell the quoted strings are turned to quoteless ones

Escaping may be necessary

**Reply by mandras73** [reply]
Thank you Gents, escaping the run configuration did the trick

`-m=unit(state=\"new\")`
